Indicators/ChandeKrollStop.cs

```diff
@@ -0,0 +1,118 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the short stop and lower stop values of the Chande Kroll Stop Indicator.
+    /// It is used to determine the optimal placement of a stop-loss order.
+    /// </summary>
+    public class ChandeKrollStop : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly AverageTrueRange _atr;
+        private readonly decimal _atrMult;
+
+        private readonly Maximum _highStopMaximum;
+        private readonly Minimum _lowStopMinimum;
+
+        /// <summary>
+        /// Gets the short stop of ChandeKrollStop.
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> ShortStop { get; }
+
+        /// <summary>
+        /// Gets the long stop of ChandeKrollStop.
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> LongStop { get; }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples >= WarmUpPeriod;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChandeKrollStop"/> class.
+        /// </summary>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(int atrPeriod, decimal atrMult, int period)
+            : this($"CKS({atrPeriod},{atrMult},{period})", atrPeriod, atrMult, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChandeKrollStop"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(string name, int atrPeriod, decimal atrMult, int period)
+            : base(name)
+        {
+            WarmUpPeriod = atrPeriod + period - 1;
+
+            _atr = new AverageTrueRange(atrPeriod);
```

Comment:
Let's add the optional `MovingAverageType` for `ChandeKrollStop` to be passed to the ATR 👍

Reply:
Done